File: app/src/main/java/is/hello/parallax/InsightCard.java
package is.hello.parallax;

import android.content.res.Resources;
import android.graphics.drawable.Drawable;
import android.support.annotation.DrawableRes;
import android.support.annotation.NonNull;
import android.support.annotation.StringRes;
import android.support.v4.content.res.ResourcesCompat;

public class InsightCard {
    public final Drawable drawable;
    public final String title;

    public InsightCard(Drawable drawable, String title) {
        this.drawable = drawable;
        this.title = title;
    }

    public InsightCard(@NonNull Resources resources,
                       @DrawableRes int drawableRes,
                       @StringRes int titleRes) {
        this(ResourcesCompat.getDrawable(resources, drawableRes, null),
             resources.getString(titleRes));
    }

    @Override
    public String toString() {
        return "InsightCard{" +
                "drawable=" + drawable +
                ", title='" + title + '\'' +
                '}';
    }
}

File: app/src/main/kotlin/is/hello/parallax/InsightCard.kt
package `is`.hello.parallax

import android.content.res.Resources
import android.graphics.drawable.Drawable
import android.support.annotation.DrawableRes
import android.support.annotation.StringRes
import android.support.v4.content.res.ResourcesCompat

data class InsightCard(val drawable: Drawable, val title: String) {
    constructor(resources: Resources,
                @DrawableRes drawableRes: Int,
                @StringRes titleRes: Int)
        : this(ResourcesCompat.getDrawable(resources, drawableRes, null),
               resources.getString(titleRes)) {}
}
